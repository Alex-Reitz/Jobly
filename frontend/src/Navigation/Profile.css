.logged-in-warning {
  color: black;
  text-align: center;
  font-size: 2rem;
  margin: 0;
  position: absolute;
  top: 50%;
  left: 50%;
  text-transform: capitalize;
  transform: translate(-50%, -50%);
}
#update {
  background-color: #316685;
  border: none;
  color: white;
  padding: 10px;
  text-decoration: none;
  margin: 4px 2px;
  font-size: 1rem;
  cursor: pointer;
  border-radius: 5px;
}
.profile-heading {
  margin: 10px;
  text-transform: capitalize;
  padding-top: 50px;
}
.profile-input {
  width: 15%;
  padding: 0.25em;
}
input:focus {
  border: 3px solid #316685;
}
.profile-div {
  margin: 5px;
}
.profile-label {
  margin: 3px;
  display: inline-block;
  width: 100px;
}
.Link {
  background-color: #316685;
  border: none;
  color: white;
  padding: 10px;
  text-decoration: none;
  margin: 4px 2px;
  font-size: 1rem;
  cursor: pointer;
  border-radius: 5px;
}
