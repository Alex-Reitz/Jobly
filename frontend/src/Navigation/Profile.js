import React, { useContext, useState } from "react";
import UserContext from "../Context/userContext";
import { Link } from "react-router-dom";
import JoblyApi from "../api";
import "./Profile.css";

function Profile() {
  const { currentUser } = useContext(UserContext);
  //sets the form state
  const [formData, setFormData] = useState({
    password: "",
    firstName: "",
    lastName: "",
    email: "",
  });
  //updates state based on change in form
  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };
  //Sends the patch request out to backend
  const gatherInput = (evt) => {
    evt.preventDefault();
    console.log({ ...formData });
    update(currentUser.username, { ...formData });
    setFormData({
      firstName: "",
      lastName: "",
      email: "",
      password: "",
    });
  };
  //sends update to backend
  const update = (username, data) => {
    async function updateUser(username, data) {
      const res = await JoblyApi.update(username, data);
      return res;
    }
    updateUser(username, data);
  };

  if (!currentUser.username) {
    return (
      <div className="logged-in-warning">
        <h3>Login to see your profile</h3>
        <Link className="Link" to="/signup">
          Signup
        </Link>
        <Link className="Link" to="/login">
          Login
        </Link>
      </div>
    );
  } else {
    return (
      <div className="profile-content">
        <h4 className="profile-heading">
          Update your profile here, {currentUser.username}
        </h4>
        <form onSubmit={gatherInput}>
          <div className="profile-div">
            <label className="profile-label" htmlFor="firstName">
              First Name
            </label>
            <input
              className="profile-input"
              placeholder={currentUser.firstName}
              onChange={handleChange}
              type="text"
              name="firstName"
              value={formData.firstName}
              id="firstName"
            />
          </div>
          <div className="profile-div">
            <label className="profile-label" htmlFor="lastName">
              Last Name
            </label>
            <input
              className="profile-input"
              placeholder={currentUser.lastName}
              onChange={handleChange}
              type="text"
              name="lastName"
              value={formData.lastName}
              id="lastName"
            />
          </div>
          <div className="profile-div">
            <label className="profile-label" htmlFor="email">
              Email
            </label>
            <input
              className="profile-input"
              placeholder={currentUser.email}
              onChange={handleChange}
              type="text"
              name="email"
              value={formData.email}
              id="email"
            />
          </div>
          <div className="profile-div">
            <label className="profile-label" htmlFor="password">
              Password
            </label>
            <input
              className="profile-input"
              onChange={handleChange}
              type="password"
              name="password"
              placeholder="Password To Confirm Update"
              value={formData.password}
              id="password"
            />
          </div>
          <button id="update">Update</button>
        </form>
      </div>
    );
  }
}

export default Profile;
