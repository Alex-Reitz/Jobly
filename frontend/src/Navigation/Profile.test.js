import React from "react";
import { render } from "@testing-library/react";
import Profile from "./Profile";
import { MemoryRouter } from "react-router";
import UserContext from "../Context/userContext";

it("matches snapshot", function () {
  const testFunc = jest.fn();
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser: testFunc }}>
        <Profile />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
