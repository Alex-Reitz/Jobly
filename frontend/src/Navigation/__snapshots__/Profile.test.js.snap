// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="logged-in-warning"
  >
    <h3>
      Login to see your profile
    </h3>
    <a
      class="Link"
      href="/signup"
    >
      Signup
    </a>
    <a
      class="Link"
      href="/login"
    >
      Login
    </a>
  </div>
</DocumentFragment>
`;
